File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m * n != len(original):
            return []
        res = [[None for _ in range(0, n)] for _ in range(0, m)]
        i = 0
        for r in range(0, m):
            for c in range(0, n):
                res[r][c] = original[i]
                i += 1
        return res

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.js
/**
 * @param {number[]} original
 * @param {number} m
 * @param {number} n
 * @return {number[][]}
 */
var construct2DArray = function(original, rows, cols) {
    if (rows * cols !== original.length) return []
    const res = Array(rows).fill(0).map(() => Array(cols).fill(0).map(() => ([])))
    let i = 0
    for (let r=0; r < rows; r++) {
        for (let c=0; c < cols; c++) {
            res[r][c] = original[i++]
        }
    }
    return res
};